safe/gui/widgets/dock.py

```diff
@@ -1075,13 +1080,18 @@ def generate_impact_report(self, impact_function):
         impact_report = ImpactReportV4(
             self.iface,
             report_metadata,
-            impact_function=impact_function)
+            impact_function=impact_function,
+            minimum_needs_profile=minimum_needs)
 
         # generate report folder
 
         # no other option for now
         # TODO: retrieve the information from data store
-        layer_dir = os.path.dirname(impact_function.impact.source())
+        if isinstance(impact_function.datastore.uri, QDir):
+            layer_dir = impact_function.datastore.uri.absolutePath()
+        else:
+            # No other way for now
+            return
```

Comment:
Store the generated impact report in the folder datastore.
In the future, we will replace it by generating the report directly and take all the information from datastore API.